Generator/Templates/native.safehandle.sbntxt

```diff
@@ -46,7 +46,7 @@ namespace {{ namespace.native_name }}
                 else
                 {
                     var str = new {{symbol_name}}();
-                    Marshal.StructureToPtr(str, ptr, true);
+                    Marshal.StructureToPtr(str, ptr, false);
```

Comment:
Did this change fix some error for you?

Reply:
Changing this (in addition to stopping usage of `ref safehandle`) fixed getting `AccessViolationException` and other heap corruption related problems.

Reading the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.structuretoptr?view=net-5.0), I get the impression it should indeed be false:

> **fDeleteOld**
> true to call the DestroyStructure(IntPtr, Type) method on the ptr parameter before this method copies the data. The block must contain valid data. Note that passing false when the memory block already contains data can lead to a memory leak.

As we create the (blank) pointer above with `AllocHGlobal`.

Reply:
Isn't it the other way around? We create a pointer to some memory location. What if this memory location still holds old data which was not cleared. But I could be wrong about this.

Perhaps it is working again with my other comment regarding `caller-allocates`?

We probably should understand what is the correct way of doing it and why, add a comment and be done with it.

One of us could ask on gitter in #csharplang. I made very good experiences there. The people are quite helpful.